asyncstdlib/asynctools.py

```diff
@@ -174,3 +177,31 @@ async def head_tail(iterable, leading=5, trailing=5):
     if not hasattr(iterator, "aclose"):
         return nullcontext(iterator)
     return _ScopedAsyncIteratorContext(iterator)
+
+
+async def await_each(awaitables: Iterable[Awaitable[T]]) -> AsyncIterable[T]:
+    """
+    Iterate through ``awaitables`` and await each item
+
+    This converts an *iterable of async* into an *async iterator* of awaited values.
+    Consequently, we can apply various functions made for ``AsyncIterable[T]`` to
+    ``Iterable[Awaitable[T]]`` as well.
+
+    This is particularly practical when you need to make up for the lack of async
+    lambdas in Python, where your lambda needs to return a single coroutine.
+
+    For example:
+
+    .. code-block:: python3
+
+        import asyncstdlib as a
+
+         async def async_is_non_negative(x: int) -> bool:
+              return x > 0
+
+         all_non_negative = await a.all(
+             a.await_each(
+                 async_is_non_negative(x) for x in [1, 2, 3]))
```

Comment:
I'm somewhat unhappy with this example. It's not really clear how "lack of lambda functions" relates to the code.
It is perfectly possible to do
```python3
await a.all(await async_is_non_negative(x) for x in [1, 2, 3])
await a.all(a.map(await async_is_non_negative, [1, 2, 3]))
```

Perhaps just implying there are some pre-built awaitables would do the trick:
```python3
okay = await a.all(a.await_each([check_a, check_b, check_c]))
```

Reply:
I re-read and you are right, this is a confusing example. Would putting icontract be too much for the example? Maybe you have a better idea? Any other library or operation?

Reply:
Actually, I'll leave out mentioning the lack of lambda functions and just give your example with the checks, where checks are functions returning booleans.